src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1319,6 +1366,431 @@ CreateCitusTable(Oid relationId, CitusTableType tableType,
 }
 
 
+/*
+ * ConvertCitusLocalTableToTableType converts given Citus local table to
+ * given table type.
+ *
+ * This only supports converting Citus local tables to reference tables
+ * (by replicating the shard to workers) and single-shard distributed
+ * tables (by replicating the shard to the appropriate worker and dropping
+ * the local one).
+ */
+static void
+ConvertCitusLocalTableToTableType(Oid relationId, CitusTableType tableType,
```

Comment:
can we move some of this logic into a different file? this one is getting kind of huge